### PR TITLE
Move cuda instantiations in a plugin

### PR DESCRIPTION
**Pull request by alxbilger**
The design has been decided here: https://github.com/sofa-framework/sofa/pull/4395
Related: https://github.com/SofaDefrost/SoftRobots/pull/259

### CODE REVIEW

**Review by fredroy — APPROVED** [review]

**Review by bakpaul — APPROVED** [review]

File: CMakeLists.txt [hunk 1]
```diff
@@ -196,4 +184,6 @@ SET_PROPERTY(TARGET ${PROJECT_NAME} PROPERTY FOLDER "plugins")
 # Config files and install rules for pythons scripts
 sofa_install_pythonscripts(PLUGIN_NAME ${PROJECT_NAME} PYTHONSCRIPTS_SOURCE_DIR "python")
 
+sofa_add_subdirectory(plugin extensions/CUDA BeamAdapter.CUDA)
```
Comment (epernod) [inline CMakeLists.txt:187]: There should no be a if(SofaCUDA_FOUND) here?
What is the behavior if SofaCUDA is not compiled?

Comment (alxbilger) [reply CMakeLists.txt:187]: This line does not activate the compilation of the plugin. As any plugin, the developer must enable it in cmake.

Comment (epernod) [reply CMakeLists.txt:187]: ok so default value is OFF?

Comment (epernod) [reply CMakeLists.txt:187]: and if it is set to ON and SofaCUDA is not present, we will have a cmake error, right?

Comment (alxbilger) [reply CMakeLists.txt:187]: It is OFF by default. And you are right about the cmake error.

**Review by epernod — APPROVED** [review]

**Pull request merged**